Sources/PackageSets/API.swift

```diff
@@ -0,0 +1,179 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2020 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import PackageModel
+import SourceControl
+
+public protocol PackageSetsProtocol {
+    // MARK: - Package set profile APIs
+
+    /// Lists package set profiles.
+    ///
+    /// The result of this API does not include `PackageSet` data. All other APIs in this
+    /// protocol require the context of a profile. Implementations should support a "default"
+    /// profile such that the `profile` API parameter can be optional.
+    ///
+    /// - Parameters:
+    ///   - callback: The closure to invoke when result becomes available
+    func listProfiles(callback: @escaping (Result<[PackageSetsModel.Profile], Error>) -> Void)
+
+    // MARK: - Package set APIs
+
+    /// Returns packages organized into groups.
+    ///
+    /// Package sets are not mutually exclusive; a package may belong to more than one group. As such,
+    /// the ordering of `PackageSet`s should be preserved and respected during conflict resolution.
+    ///
+    /// - Parameters:
+    ///   - identifers: Optional. If specified, only `PackageSet`s with matching identifiers will be returned.
+    ///   - profile: Optional. The `PackageSetProfile` context. By default the `default` profile is used.
+    ///   - callback: The closure to invoke when result becomes available
+    func listPackageSets(
+        identifers: Set<PackageSetsModel.PackageSetIdentifier>?,
+        in profile: PackageSetsModel.Profile?,
+        callback: @escaping (Result<[PackageSetsModel.PackageSet], Error>) -> Void
+    )
+
+    /// Refreshes all configured package sets.
+    ///
+    /// - Parameters:
+    ///   - profile: Optional. The `PackageSetProfile` context. By default the `default` profile is used.
+    ///   - callback: The closure to invoke after triggering a refresh for the configured package sets.
+    func refreshPackageSets(
+        in profile: PackageSetsModel.Profile?,
+        callback: @escaping (Result<[PackageSetsModel.PackageSetSource], Error>) -> Void
+    )
+
+    /// Adds a package set.
+    ///
+    /// - Parameters:
+    ///   - source: The package set's source
+    ///   - order: Optional. The order that the `PackageSet` should take after being added to the list.
+    ///            By default the new group is appended to the end (i.e., the least relevant order).
```

Comment:
nit: should this say "set" instead of "group"?

Reply:
renamed this to collections and addressed